scraper/src/mindtouch2zim/client.py

```diff
@@ -307,32 +300,41 @@ def get_page_content(self, page: LibraryPage) -> LibraryPageContent:
             )
         return LibraryPageContent(html_body=tree["body"][0])
 
-    def get_page_definition(self, page: LibraryPage) -> LibraryPageDefinition:
+    def get_page_definition(self, page: LibraryPage | str) -> LibraryPageDefinition:
         """Return the definition of a given page
 
         Definition is kept in memory, and retrieved on-demand when it is not yet there
         """
-        if page.definition is None:
-            raw_definition = self._get_api_json(
-                f"/pages/{page.id}", timeout=CONTEXT.http_timeout_normal_seconds
-            )
-            raw_tags = raw_definition.get("tags", None)
-            if raw_tags is None:
-                raise MindtouchParsingError(f"No tags property for page {page.id}")
-            raw_tag = raw_tags.get("tag", None)
-            if raw_tag is None:
-                raise MindtouchParsingError(f"No tag property for page {page.id}")
-            if isinstance(raw_tag, list):
-                tags = [item.get("@value") for item in raw_tag]
-            else:
-                tags = [raw_tag.get("@value")]
-            page.definition = LibraryPageDefinition(
-                tags=tags,
-            )
-        return page.definition
+        if isinstance(page, str):
+            page_id = page
+        elif page.definition is not None:
+            return page.definition
+        else:
+            page_id = page.id
+
+        raw_definition = self._get_api_json(
+            f"/pages/{page_id}", timeout=CONTEXT.http_timeout_normal_seconds
+        )
+        raw_tags = raw_definition.get("tags", None)
+        if raw_tags is None:
+            raise MindtouchParsingError(f"No tags property for page {page_id}")
+        raw_tag = raw_tags.get("tag", None)
+        if raw_tag is None:
```

Comment:
Cant we combine those two conditions into one?

Reply:
Indeed, done